### PR TITLE
Mekanist DZ60rgb Keymap

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

Checking in @nshanpetrosyan (mekanist) keymap so he can retain all his functionality and easily edit it on his own. Special thanks to @MatthewRobo for helping create the layer rgb lightings. 

I've included a helpful `readme.md` as well. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by race41989** [comment]
Is this a potential fix to thw Configurator  compile error?

**Comment by mechmerlin** [comment]
> Is this a potential fix to the Configurator compile error?

tl;dr: @race41989 o it is not. This is a keymap for mekanist's keyboard that has a bunch of his features that I've been helping him with. Mekanist is a member of the community who I've been chatting with for awhile and I figured I'd help him do this. 

Creating a firmware file for your keyboard can be done in 2 different ways. The first is via the online GUI known as QMK Configurator. The second way is to compile the source code yourself with the keymap and functionality you want. This PR was specifically made to assist mekanist in compiling the source himself with his keymap and desired functionality. It has nothing to do with the breakage in QMK Configurator. 

While the QMK Configurator is broken for dz60rgb, you can do what mekanist is doing here as well. You can consult our [guide](https://beta.docs.qmk.fm/newbs) on how to setup a development environment and how to create your keymap from scratch.

**Comment by nshanpetrosyan** [comment]
Thanks again for your help @mechmerlin !

**Comment by drashna** [comment]
Thanks!
